google/cloud/security/common/gcp_api/_base_repository.py

```diff
@@ -0,0 +1,699 @@
+    # Lower the default logging level of just this module to WARNING.
+    logging.getLogger(discovery.__name__).setLevel(logging.WARNING)
+
+    discovery_kwargs = {
+        'serviceName': service_name,
+        'version': version,
+        'developerKey': developer_key,
+        'credentials': credentials}
+    if SUPPORT_DISCOVERY_CACHE:
+        discovery_kwargs['cache_discovery'] = cache_discovery
+
+    return discovery.build(**discovery_kwargs)
+
+
+def _set_user_agent(credentials):
+    """Set custom Forseti user agent for all authenticated requests.
+
+    Args:
+        credentials (OAuth2Credentials): The credentials object used to
+            authenticate all http requests.
+    """
+    if isinstance(credentials, client.OAuth2Credentials):
+        user_agent = credentials.user_agent
+        if (not user_agent or
+                forseti_security.__package_name__ not in user_agent):
+
+            credentials.user_agent = (
+                'Python-httplib2/{} (gzip), {}/{}'.format(
+                    httplib2.__version__,
+                    forseti_security.__package_name__,
+                    forseti_security.__version__))
+
+
+def credential_from_keyfile(keyfile_name, scopes, delegated_account):
+    """Build delegated credentials required for accessing the gsuite APIs.
+
+    Args:
+        keyfile_name (str): The filename to load the json service account key
+            from.
+        scopes (list): The list of required scopes for the service account.
+        delegated_account (str): The account to delegate the service account to
+            use.
+
+    Returns:
+        OAuth2Credentials: Credentials as built by oauth2client.
+
+    Raises:
+        api_errors.ApiExecutionError: If fails to build credentials.
+    """
+    try:
+        credentials = (
+            service_account.ServiceAccountCredentials.from_json_keyfile_name(
+                keyfile_name, scopes=scopes))
+    except (ValueError, KeyError, TypeError, IOError) as e:
+        raise api_errors.ApiExecutionError(
+            'Error building admin api credential: %s', e)
+
+    return credentials.create_delegated(delegated_account)
+
+
+def flatten_list_results(paged_results, item_key):
+    """Flatten a split-up list as returned by list_next() API.
+
+    GCE 'list' APIs return results in the form:
+      {item_key: [...]}
+    with one dictionary for each "page" of results. This method flattens
+    that to a simple list of items.
+
+    Args:
+        paged_results (list): A list of paged API response objects.
+            [{page 1 results}, {page 2 results}, {page 3 results}, ...]
+        item_key (str): The name of the key within the inner "items" lists
+            containing the objects of interest.
+
+    Returns:
+        list: A list of items.
+    """
+    results = []
+    for page in paged_results:
+        results.extend(page.get(item_key, []))
+    return results
```

Comment:
super-nit: Sometimes you leave a new line before the return, other time's you don't. Consistency is probably good here.